seurat-find-neighbours.R

```diff
@@ -0,0 +1,208 @@
+    default = "euclidean",
+    type = 'character',
+    help = "Distance metric for annoy. Options include: euclidean (default), cosine, manhattan, and hamming"
+  ),
+  make_option(
+    c("--graph-name"),
+    action = "store",
+    default = NULL,
+    type = 'character',
+    help = "Name of graph to use for the clustering algorithm."
+  ),
+  make_option(
+    c("--nn-eps"),
+    action = "store",
+    default = 0.0,
+    type = 'double',
+    help = "Error bound when performing nearest neighbor seach using RANN; default of 0.0 implies exact nearest neighbor search"
+  ),
+  make_option(
+    c("--verbose"),
+    action = "store_true",
+    default = FALSE,
+    help = "Maximal number of iterations per random start"
+  ),
+  make_option(
+    c("--force-recalc"),
+    action = "store_true",
+    default = FALSE,
+    help = "Force recalculation of SNN"
+  ),
+  make_option(
+    c("-f", "--features"),
+    action = "store",
+    default = NULL,
+    type = 'character',
+    help = "Comma-separated list of genes to use for building SNN. Alternatively, text file with one gene per line."
+  ),
+  make_option(
+    c("--reduction"),
+    action = "store",
+    default = NULL,
+    type = 'character',
+    help = "Reduction to use as input for building the SNN"
+  ),
+  make_option(
+    c("--dims"),
+    action = "store",
+    default = NULL,
+    type = 'character',
+    help = "Dimensions of reduction to use as input. A comma-separated list of the dimensions to use in construction of the SNN graph (e.g. To use the first 5 PCs, pass 1,2,3,4,5)."
+  ),
+  make_option(
+    c("--assay"),
+    action = "store",
+    default = NULL,
+    type = 'character',
+    help = "Assay to use in construction of SNN"
+  ),
+  make_option(
+    c("--do-plot"),
+    action = "store_true",
+    default = FALSE,
+    help = "Plot SNN graph on tSNE coordinates"
+  )
+)
+
+opt <- wsc_parse_args(option_list, mandatory = c('input_object_file', 'output_object_file'))
+
+# Check parameter values
+
+if ( ! file.exists(opt$input_object_file)){
+  stop((paste('File', opt$input_object_file, 'does not exist')))
+}
+
+dims <- opt$dims
+if ( ! is.null(dims)){
+  dims <- as.integer(wsc_parse_numeric(opt, 'dims'))
+}
+
+features <- NULL
+if (! is.null(opt$features)){
+  if (! file.exists(opt$features)){
+    stop((paste('Supplied genes file', opt$features, 'does not exist')))
+  }else{
+    features <- readLines(opt$features)
+  }
+}
+
+# Now we're hapy with the arguments, load Seurat and do the work
+
+suppressPackageStartupMessages(require(Seurat))
+if(opt$input_format == "loom" | opt$output_format == "loom") {
+  suppressPackageStartupMessages(require(loomR))
+} else if(opt$input_format == "singlecellexperiment" | opt$output_format == "singlecellexperiment") {
+  suppressPackageStartupMessages(require(scater))
+}
+
+# Input from serialized R object
+
+seurat_object <- read_seurat3_object(input_path = opt$input_object_file, format = opt$input_format)
+
```

Comment:
We need to open a PDF/PNG renderer if the plotting is activated, before and after the method call.